### PR TITLE
test(functional): validate pods dnsPolicy

### PR DESCRIPTION
**Pull request by juliayakovlev**
Task:
https://trello.com/c/OkmLsxou/3579-check-that-clusterfirstwithhostnet-dnspolicy-is-used-by-
scylla-operator

Check that 'ClusterFirstWithHostNet' DNSPolicy is used by Scylla cluster

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: functional_tests/scylla_operator/test_functional.py [hunk 1]
```diff
@@ -581,3 +581,13 @@ def test_scylla_yaml_override(db_cluster, scylla_yaml):  # pylint: disable=too-m
 
     with scylla_yaml() as props:
         assert configmap_scylla_yaml_content == props
+
+
+@pytest.mark.readonly
+def test_default_dns_policy(db_cluster: ScyllaPodCluster):
+    pods = db_cluster.k8s_cluster.kubectl("get pods -l scylla/cluster=sct-cluster -o yaml", namespace=SCYLLA_NAMESPACE)
```
Comment (vponomaryov) [inline functional_tests/scylla_operator/test_functional.py:588]: "sct-cluster" should not be used directly. It is just default value of the "k8s_scylla_cluster_name" config option. Use config options only to get the data defined by it.

File: functional_tests/scylla_operator/test_functional.py [hunk 2]
```diff
@@ -581,3 +581,13 @@ def test_scylla_yaml_override(db_cluster, scylla_yaml):  # pylint: disable=too-m
...
+
+    dns_policy = set(pod["spec"]["dnsPolicy"] for pod in yaml.safe_load(pods.stdout)["items"] if pod)
+
+    assert dns_policy == {'ClusterFirstWithHostNet', }, (
+        f'Expected "ClusterFirstWithHostNet" dnsPolicy on all nodes, actually it is {dns_policy}')
```
Comment (vponomaryov) [inline functional_tests/scylla_operator/test_functional.py:593]: Strange to see possible error message with a set of strings without relation to pods. 
I think would be better to do the same as in the test with tags checking...

**Review by vponomaryov — CHANGES_REQUESTED** [review]

File: functional_tests/scylla_operator/test_functional.py (hunk 1 above)
Comment (juliayakovlev) [reply functional_tests/scylla_operator/test_functional.py]: done

File: functional_tests/scylla_operator/test_functional.py (hunk 2 above)
Comment (juliayakovlev) [reply functional_tests/scylla_operator/test_functional.py]: changed

File: functional_tests/scylla_operator/test_functional.py [hunk 3]
```diff
@@ -581,3 +581,24 @@ def test_scylla_yaml_override(db_cluster, scylla_yaml):  # pylint: disable=too-m
 
     with scylla_yaml() as props:
         assert configmap_scylla_yaml_content == props
+
+
+@pytest.mark.readonly
+def test_default_dns_policy(db_cluster: ScyllaPodCluster):
+    expected_policy = "ClusterFirstWithHostNet"
+    pods = db_cluster.k8s_cluster.kubectl(f"get pods -l scylla/cluster="
```
[the remark this answers is not in the document]
Comment (juliayakovlev) [reply functional_tests/scylla_operator/test_functional.py]: removed

**Review by vponomaryov — APPROVED** [review]

**Pull request merged**